pkg/controller/trustpolicyexception_api.go

```diff
@@ -26,28 +26,30 @@ import (
 )
 
 type TrustPolicyExceptionApi struct {
-	all   *AllApis
-	sync  *regiondata.Sync
-	store edgeproto.TrustPolicyExceptionStore
-	cache edgeproto.TrustPolicyExceptionCache
+	all       *AllApis
+	sync      *regiondata.Sync
+	store     edgeproto.TrustPolicyExceptionStore
+	cache     edgeproto.TrustPolicyExceptionCache
+	instStore edgeproto.TPEInstanceStateStore
+	instCache edgeproto.TPEInstanceStateCache
 }
 
 func NewTrustPolicyExceptionApi(sync *regiondata.Sync, all *AllApis) *TrustPolicyExceptionApi {
 	trustPolicyExceptionApi := TrustPolicyExceptionApi{}
 	trustPolicyExceptionApi.all = all
 	trustPolicyExceptionApi.sync = sync
 	trustPolicyExceptionApi.store = edgeproto.NewTrustPolicyExceptionStore(sync.GetKVStore())
+	trustPolicyExceptionApi.instStore = edgeproto.NewTPEInstanceStateStore(sync.GetKVStore())
 	edgeproto.InitTrustPolicyExceptionCache(&trustPolicyExceptionApi.cache)
+	edgeproto.InitTPEInstanceStateCache(&trustPolicyExceptionApi.instCache)
 	sync.RegisterCache(&trustPolicyExceptionApi.cache)
+	sync.RegisterCache(&trustPolicyExceptionApi.instCache)
 	return &trustPolicyExceptionApi
 }
 
 func (s *TrustPolicyExceptionApi) CreateTrustPolicyException(ctx context.Context, in *edgeproto.TrustPolicyException) (*edgeproto.Result, error) {
 
 	log.SpanLog(ctx, log.DebugLevelApi, "CreateTrustPolicyException", "policy", in)
-	if len(in.OutboundSecurityRules) == 0 {
```

Comment:
why is this check not required anymore?

Reply:
I moved that check into the in.Validate() function, so it's still there. The problem was the Update function wasn't checking if the user had removed all the rules, so now Validate() checks for that on both Create and Update.